### ISSUE TITLE
Page object being Memory Leaked inside PuppeteerExtraPluginAdblocker

### ISSUE THREAD

**Issue by victornpb** [berstend/puppeteer-extra#375]
## The issue

The memory leak occurs inside the [enableBlockingInPage()](https://github.com/cliqz-oss/adblocker/blob/1a8863652a9ccda02ab1578af8166e1b48018e84/packages/adblocker-playwright/adblocker.ts#L107) method from the PuppeteerBlocker library. It [holds a reference to every page](https://github.com/cliqz-oss/adblocker/blob/1a8863652a9ccda02ab1578af8166e1b48018e84/packages/adblocker-playwright/adblocker.ts#L119
)  in a [Map](https://github.com/cliqz-oss/adblocker/blob/1a8863652a9ccda02ab1578af8166e1b48018e84/packages/adblocker-playwright/adblocker.ts#L106).

![image](https://user-images.githubusercontent.com/3372598/99575725-c8d3a680-29b7-11eb-85e4-a4e20be1281e.png)

## Reproduction

Minimum reproduction case:

- Launch browser
- Repeat 100 times:
  - Create a page;
  - Close page; 

```js
// const puppeteer = require('puppeteer');
const puppeteer = require('puppeteer-extra');
const AdblockerPlugin = require('puppeteer-extra-plugin-adblocker');

puppeteer.use(AdblockerPlugin());

const delay = ms => new Promise(r => setTimeout(r, ms));

(async () => {
    console.log('Start');

    const browser = await puppeteer.launch({
        headless:false,
    });
    
    const p = [];
    for (let i = 0; i < 100; i++) p.push(createPage(browser, 1))

    await Promise.all(p);

    browser.close();
    console.log('End!');
})();

async function createPage(browser, t) {
    const page = await browser.newPage();
    await page.goto('about:blank');
    await delay(t);
    await page.close();
}
```

## Profiling 
![image](https://user-images.githubusercontent.com/3372598/99575547-8742fb80-29b7-11eb-8f6f-ddefecd98d2b.png)

[PuppeteerExtraPluginAdblocker](https://github.com/berstend/puppeteer-extra/blob/master/packages/puppeteer-extra-plugin-adblocker/src/index.ts) has this method [onPageCreated()](https://github.com/berstend/puppeteer-extra/blob/master/packages/puppeteer-extra-plugin-adblocker/src/index.ts#L129) that is called every time a page is created.
```js
async onPageCreated(page: any) {
    this.debug('onPageCreated')
    ;(await this.getBlocker()).enableBlockingInPage(page)
  }
```

## Solution

Looking at the code, I can see that there's a method called [disableBlockingInPage](https://github.com/cliqz-oss/adblocker/blob/1a8863652a9ccda02ab1578af8166e1b48018e84/packages/adblocker-playwright/adblocker.ts#L119), which [removes](https://github.com/cliqz-oss/adblocker/blob/1a8863652a9ccda02ab1578af8166e1b48018e84/packages/adblocker-playwright/adblocker.ts#L125) that reference from the [map](https://github.com/cliqz-oss/adblocker/blob/1a8863652a9ccda02ab1578af8166e1b48018e84/packages/adblocker-playwright/adblocker.ts#L101).

It looks like this method should be called automatically on "page.close" or exposed so the user can manage it by themselves.

I can confirm that manually calling the disableBlockingInPage() will GarbageCollect all 100 instances of page.

Another solution that I confirmed works (at least in this minimal case) is that they could use a WeakMap instead of a Map, so It is the library consumer responsibility to hold a strong reference to the page, and not the library internals. But I don't know if they need to keep a strong reference to the page, to do book-keeping to other stuff inside there. (I'm going to[ open an issue there as well](https://github.com/cliqz-oss/adblocker/issues/1449), to understand if they have any reasons to use a Map instead of a WeakMap).

**Comment by victornpb** [comment]
Update: The adblocker pushed yesterday a fix changing the Map to a WeakMap, so now it will not have a strong reference to the pages, and is up to the consumer to retain those objects.

This effectively makes those objects Garbage Collectable and before they weren't.

https://github.com/cliqz-oss/adblocker/issues/1449#issuecomment-730234360

----

Apart from that, I still would like to see puppeteer-extra-plugin-adblocker expose the "release" methods to the puppeteer-extra consumers, so they can choose, when they want that object to be dereferenced.

**Comment by berstend** [comment]
Fix released as `puppeteer-extra-plugin-adblocker@2.11.9`
thanks @victornpb 👍
